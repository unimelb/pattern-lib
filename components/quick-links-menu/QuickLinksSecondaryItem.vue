<template>
  <a
    :href="href"
    class="ql-menu__secondary-link">
    <SvgIcon
      :name="icon"
      class="ql-menu__secondary-icon"
      height="16"
      width="16" />
    <span class="ql-menu__secondary-text">{{ title }}</span>
  </a>
</template>

<script>
import SvgIcon from '../icons/SvgIcon.vue';

export default {
  components: {
    SvgIcon,
  },
  props: {
    href: {
      type: String,
      default: '',
      required: true,
    },
    title: {
      type: String,
      default: '',
      required: true,
    },
    icon: {
      type: String,
      default: 'chevron-right',
    },
  },
};
</script>
