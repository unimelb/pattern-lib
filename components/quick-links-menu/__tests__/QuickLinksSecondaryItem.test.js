import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import QuickLinksSecondaryItem from '../QuickLinksSecondaryItem.vue';

expect.extend(toHaveNoViolations);

describe('QuickLinksSecondaryItem', () => {
  it('should match snapshot', () => {
    const result = shallow(QuickLinksSecondaryItem, {
      propsData: {
        href: 'www.unimelb.edu.au',
        title: 'mock title',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(QuickLinksSecondaryItem, {
      propsData: {
        href: 'www.unimelb.edu.au',
        title: 'mock title',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
