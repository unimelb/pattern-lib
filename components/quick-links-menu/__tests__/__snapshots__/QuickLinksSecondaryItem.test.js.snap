// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickLinksSecondaryItem should match snapshot 1`] = `
<a
  class="ql-menu__secondary-link"
  href="www.unimelb.edu.au"
>
  <!---->
   
  <span
    class="ql-menu__secondary-text"
  >
    mock title
  </span>
</a>
`;
